src/provider/provider_os_memory_internal.h

```diff
@@ -8,9 +8,10 @@
 #ifndef UMF_OS_MEMORY_PROVIDER_INTERNAL_H
 #define UMF_OS_MEMORY_PROVIDER_INTERNAL_H
 
+#include "critnib.h"
 #include "utils_common.h"
+#include <hwloc.h>
```

Comment:
I'm not sure what is the convention but I think we should always include system headers with <> first and then local